core/transaction-pool/graph/src/base_pool.rs

```diff
@@ -356,6 +366,14 @@ impl<Hash: hash::Hash + Member + Serialize, Ex: ::std::fmt::Debug> BasePool<Hash
 		removed
 	}
 
+	/// Removes ready transaction represented by the hash and all other ready transactions
+	/// that depend on it.
+	///
+	/// Returns a list of actually removed transactions.
+	pub fn remove_from_ready(&mut self, hash: &Hash) -> Vec<Arc<Transaction<Hash, Ex>>> {
```

Comment:
Well, I just wanted to avoid calling `remove_invalid()` from `resubmit()` just because transactions that `resubmit()` deals with are not necessary invalid. So thought that it is better to hide the fact that it actually uses the same algorithm as `remove_invalid()` behind the name. But will remove as you've suggested.

Reply:
Thanks! I'm just afraid that someone might call `remove_from_ready` thinking that it won't mark the transactions as invalid, and it's not really true.